functions/index.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const functions = require('firebase-functions');
+const gcs = require('@google-cloud/storage')();
+const admin = require('firebase-admin');
+const jwt = require('jsonwebtoken');
+const fs = require('fs');
+
```

Comment:
Add a high-level comment describing what this file is for and how the data moves between different locations?